presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/reloading/ReloadingResourceGroupConfigurationManager.java

```diff
@@ -244,88 +223,23 @@ public synchronized void load()
         }
     }
 
-    // Populate temporary data structures to build resource group specs and selectors from db
-    private synchronized void populateFromDbHelper(Map<Long, ResourceGroupSpecBuilder> recordMap,
-            Set<Long> rootGroupIds,
-            Map<Long, ResourceGroupIdTemplate> resourceGroupIdTemplateMap,
-            Map<Long, Set<Long>> subGroupIdsToBuild)
+    private synchronized void buildResourceGroupSpecsMap(Map<ResourceGroupIdTemplate, ResourceGroupSpec> resourceGroupSpecs, List<ResourceGroupSpec> specsList, Optional<ResourceGroupIdTemplate> parentId)
```

Comment:
nit: `specsList` -> `childResourceGroups` - we don't put the data structure type in the name because we can easily figure out the data structure type in the IDE itself